src/lib/core/fiber.c

```diff
@@ -1489,19 +1500,17 @@ fiber_stack_create(struct fiber *fiber, const struct fiber_attr *fiber_attr,
 		 * after protected page until end of memory chunk can be
 		 * used for coro stack usage.
 		 */
-		guard = page_align_up(slab_data(fiber->stack_slab));
+		guard = page_align_up(stack_data_start);
 		fiber->stack = guard + page_size;
-		fiber->stack_size = slab_data(fiber->stack_slab) + stack_size -
-				    fiber->stack;
+		fiber->stack_size = stack_data_end - fiber->stack;
```

Comment:
I guess the issue is with ASAN version of small slab cache allocator. With not ASAN version everything is at least page aligned already. Let's just change `SMALL_ARENA_SLAB_ALIGNMENT` to `sizeof(long double)`. Looks like `sizeof(long long)` was a mistake (probably I thought it is 16 bytes).

Reply:
+1 for fixing it in `small`, because it violates the x86_64 Linux ABI.

But `sizeof(long double)` looks confusing (it is not guaranteed to be 128-bit), let's do:
```c
enum {
	/** Alignment for slabs from ASAN slab cache. */
	SMALL_CACHE_SLAB_ALIGNMENT = 16,
};
```

Reply:
Please keep in mind that the size provided by `fiber_attr->stack_size` might still be misaligned. A proper fix should ensure that the final result is properly aligned, so we should either:
* round `fiber_attr->stack_size` / `stack_size`,
* or still use `slab_capacity(...)` for the stack size (as I did, because otherwise we have wasted allocation space), meaning that it should respect the stack alignment.

Reply:
AFAIU in case of `stack_direction < 0` `guard`, `fiber->stack` and `fiber->stack_size` are all page aligned in case of non-ASAN build and `SMALL_CACHE_SLAB_ALIGNMENT` aligned in case of ASAN build.

Anyway do we need to care about whether stack size is 16-bytes aligned or not?

Looks like there is an issue with alignment in case of `stack_direction > 0`. We can change to  `fiber->stack = fiber->stack_slab + page_size`. It will be on par with the other direction it terms of waste.